test/unittests.cpp

```diff
@@ -738,6 +840,35 @@ void TestG2Serialization()
     }
 }
 
+void TestG2SerializationGarbage() {
+    array<uint8_t, 288> buf;
+    buf.fill(0xff);
```

Comment:
Instead of just testing only with all `0xff`, we could also loop over a large number (10k?) of  randomly generated values?

Reply:
This test is meant to test corner cases. There's another test testing random inputs.